Origin: upstream https://github.com/apache/trafficserver/pull/6767
Description: Add ts.server_response.get_maxage() (Bug: T251537)

commit 9febd97e9ad752a1dc327a2739ab1f7bfd3ff6e7
Author: Emanuele Rocca <ema@wikimedia.org>
Date:   Wed May 13 10:09:26 2020 +0200

    Lua plugin: add ts.server_response.get_maxage()

--- a/plugins/lua/ts_lua_server_response.c
+++ b/plugins/lua/ts_lua_server_response.c
@@ -44,6 +44,7 @@ static int ts_lua_server_response_get_ve
 static int ts_lua_server_response_set_version(lua_State *L);
 
 static int ts_lua_server_response_is_cacheable(lua_State *L);
+static int ts_lua_server_response_get_maxage(lua_State *L);
 
 void
 ts_lua_inject_server_response_api(lua_State *L)
@@ -155,6 +156,8 @@ ts_lua_inject_server_response_misc_api(l
 
   lua_pushcfunction(L, ts_lua_server_response_is_cacheable);
   lua_setfield(L, -2, "is_cacheable");
+  lua_pushcfunction(L, ts_lua_server_response_get_maxage);
+  lua_setfield(L, -2, "get_maxage");
 }
 
 static int
@@ -380,3 +383,17 @@ ts_lua_server_response_is_cacheable(lua_
 
   return 1;
 }
+
+static int
+ts_lua_server_response_get_maxage(lua_State *L)
+{
+  ts_lua_http_ctx *http_ctx;
+
+  GET_HTTP_CONTEXT(http_ctx, L);
+
+  TS_LUA_CHECK_SERVER_RESPONSE_HDR(http_ctx);
+
+  lua_pushnumber(L, TSHttpTxnGetMaxAge(http_ctx->txnp, http_ctx->server_response_bufp));
+
+  return 1;
+}
